src/Deployment.php

```diff
@@ -169,35 +202,52 @@ protected function isTerminalMode() {
 		return empty($_SERVER["HTTP_USER_AGENT"]);
 	}
 
+	protected static function shouldStartDeploy($config) {
+		$key = isset($config['key']) && !empty($config['key']) ? $config['key'] : self::DEFAULT_KEY;
+		return isset($_GET[$key]);
+	}
+
 	/**
+	 * Run this function in bootstrap.php before autoload.php
 	 * Install all packages/dependencies and clear cache
 	 * @param array $config
 	 */
-	public function run($config = []) {
+	public function runBase($config = []) {
+		if (!static::shouldStartDeploy($config)) {
+			return;
+		}
+
 		ob_start();
 
 		putenv('PATH=' . system('echo $PATH')); // Bez tohoto nefunguje composer (ENOGIT), protože nefunguje `which git`, protože v php.ini není PATH nastavena.
 
 		$this->installComposerDeps();
 		$this->installBowerDeps();
 
-		// clear tempDir
-		if (isset($config['tempDir']) && is_dir($config['tempDir'])) {
-			$this->removeDirectory($config['tempDir']);
+		$this->clearCache($config);
+		$this->resetAPCandOPCache();
 
-			if (Finder::findFiles('*')->exclude('.gitignore')->in($config['tempDir'])->count()) {
-				$this->log("Temp dir <bgRed>was not cleared properly<reset>.");
-			} else {
-				$this->log("Temp dir <bgGreen>cleared<reset>.");
-			}
+		ob_clean();
 
-		} else {
-			$this->log("Temp dir <cyan>is not defined<reset>.");
+		// die je až v run(), který se spustí po spuštění aplikace
+	}
+
+	/**
+	 * Clear cache
+	 * @param array $config
+	 */
+	public function run($config = []) {
+		if (!static::shouldStartDeploy($config)) {
```

Comment:
Tak v tom případě bych ho nenechával zadávat 2x. Config k deploymentu (jako je `key`) by se dal zadat pouze jednou a to v bootstrapu, ale config k ostatním (jako je redis) by zůstal v neonu jako je teď. Tento deployment config (`key`, ...) by se uložil do statické proměnné a pak se používal i v druhé části.